esx_service/utils/auth_api.py

```diff
@@ -491,6 +491,10 @@ def vm_in_any_tenant(vms):
 def _tenant_vm_add(name, vm_list):
     """ API to add vms for a tenant """
     logging.debug("_tenant_vm_add: name=%s vm_list=%s", name, vm_list)
+    if  name  == auth_data_const.DEFAULT_TENANT:
```

Comment:
I agree. More that that, it feels like a great use for decorators. Something like 
```python
@named_tenant
def _tenant_vm_add(name, vm_list):
 ...

```

where  named_tenants defined as 
```python
def named_tenant(func, *args):
   def not_supported():
      print("This feature is not supported for default vmgroup")
   if args[0] == auth_const.DEFAULT_TENANT:
      return not_supported
   return func
```